website/docs/docs/cloud/connect-data-platform/connect-snowflake.md

```diff
@@ -68,3 +73,18 @@ more information on configuring a Snowflake OAuth connection in dbt Cloud, pleas
 ## Configuration
 
 To learn how to optimize performance with data platform-specific configurations in dbt Cloud, refer to [Snowflake-specific configuration](/reference/resource-configs/snowflake-configs).
+
+## Troubleshooting
+<!--might need to turn this into details toggle if more troubleshooting items arise -->
+
+If you're receiving a `Could not deserialize key data` or `JWT token` error, refer to the following causes and solutions:
+
+- **Error: `Could not deserialize key data`**
+  - Possible causes &mdash; This could be because of mistakes like not copying correctly, missing dashes, or leaving out commented lines.
+  - Solution &mdash; You can either manually type the key values or copy the key from its source and paste it into a text editor to verify it before using it in dbt Cloud. 
+
+- **Error: `JWT token`**
+  - Possible causes &mdash; This could be transient issue between Snowflake and dbt Cloud. When connecting to Snowflake, dbt gets a JWT token valid for only 60 seconds. If there's no response from Snowflake within this time, you might see a `JWT token is invalid` error in dbt Cloud.
+  - Solution &mdash; This requires dbt being able to retry connections to Snowflake.<br /><br />
+  - Possible causes &mdash; The public key was not entered correctly in Snowflake.
+  - Solution &mdash; You can confirm and enter Snowflake's public key. Additionally, you can reach out to Snowflake for help or refer to this Snowflake doc for more info: [Key-Based Authentication Failed with JWT token is invalid Error](https://community.snowflake.com/s/article/Key-Based-Authentication-Failed-with-JWT-token-is-invalid-Error).
```

Comment:
Can those be updated to show that the solution is for the "possible cause" from above? Right now those look like 4 different points when they are actually 2 points, both with case and solution,

Reply:
fair point, it did look icky to me too!